test/server/cards/09.6-SD/InfernoGuardInvoker.spec.js

```diff
@@ -0,0 +1,105 @@
+                    type: 'military',
+                    attackers: [this.sereneWarrior],
+                    defenders: [this.solemnScholar]
+                });
+
+                this.player2.pass();
+
+                this.player1.clickCard(this.infernoGuardInvoker);
+                expect(this.player1).toBeAbleToSelect(this.sereneWarrior);
+                expect(this.player1).not.toBeAbleToSelect(this.solemnScholar);
+                expect(this.player1).not.toBeAbleToSelect(this.infernoGuardInvoker);
+                expect(this.player1).not.toBeAbleToSelect(this.fushicho);
+
+                this.player1.clickCard(this.sereneWarrior);
+                expect(this.sereneWarrior.isHonored).toBe(true);
+            });
+
+            it('should sacrifice the targeted character if the province breaks this conflict', function() {
+                this.initiateConflict({
+                    type: 'military',
+                    attackers: [this.sereneWarrior],
+                    defenders: [this.solemnScholar]
+                });
+
+                this.player2.pass();
+
+                this.player1.clickCard(this.infernoGuardInvoker);
+                expect(this.player1).toBeAbleToSelect(this.sereneWarrior);
+                expect(this.player1).not.toBeAbleToSelect(this.solemnScholar);
+                expect(this.player1).not.toBeAbleToSelect(this.infernoGuardInvoker);
+
+                this.player1.clickCard(this.sereneWarrior);
+                expect(this.sereneWarrior.isHonored).toBe(true);
+
+                this.player2.pass();
+                this.player1.pass();
+
+                this.player1.clickPrompt('No');
+                expect(this.getChatLogs(5)).toContain('Serene Warrior is discarded, burned to a pile of ash due to the delayed effect of Inferno Guard Invoker');
+                expect(this.sereneWarrior.location).toBe('dynasty discard pile');
+            });
+
+            it('should not discard the targeted character if the province doesn\'t break this conflict', function() {
+                this.initiateConflict({
+                    type: 'military',
+                    attackers: [this.sereneWarrior],
+                    defenders: [this.fushicho]
+                });
+
+                this.player2.pass();
+
+                this.player1.clickCard(this.infernoGuardInvoker);
+                this.player1.clickCard(this.sereneWarrior);
+                expect(this.sereneWarrior.isHonored).toBe(true);
+
+                this.player2.pass();
+                this.player1.pass();
+
+                expect(this.getChatLogs(5)).not.toContain('Serene Warrior is discarded, burned to a pile of ash due to the delayed effect of Inferno Guard Invoker');
+                this.player1.clickPrompt('Don\'t Resolve');
+
+                expect(this.sereneWarrior.location).toBe('play area');
+                expect(this.player1).toHavePrompt('Action Window');
+            });
+        });
+    });
+});
```

Comment:
Can you add a test for this ruling?

If a second delayed effect occurs at the end of the conflict (eg. Forebearer's Echoes), and it also causes a character to leave play, the first player will decide which resolves first. The second will not resolve, as the card has already left play.